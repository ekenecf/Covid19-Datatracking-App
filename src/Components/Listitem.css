li.countries,
.confirmed {
  font-size: 15px;
}

.listItem .confirmed {
  margin-top: 20%;
  margin-bottom: 10%;
}

.listItem .countries {
  margin-top: 15%;
}

.listItem li {
  list-style-type: none;
  color: #fff;
}

.listItem a {
  text-decoration: none;
}

.listItem:nth-child(odd) {
  background-color: #c97797;
  text-align: center;
}

.listItem:nth-child(even) {
  background-color: #cf5a89;
  text-align: center;
}
