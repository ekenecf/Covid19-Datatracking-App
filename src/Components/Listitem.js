import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './Listitem.css';

const Listitem = (props) => {
  const { countries, totalconfirmed } = props;
  return (
    <>
      <div className="listItem">
        <NavLink id={countries} to="/Details" state={{ country: countries }}>
          <li className="countries">{countries}</li>
          <li className="confirmed">{totalconfirmed}</li>
        </NavLink>
      </div>
    </>
  );
};

Listitem.propTypes = {
  countries: PropTypes.string.isRequired,
  totalconfirmed: PropTypes.string.isRequired,
};

export default Listitem;
