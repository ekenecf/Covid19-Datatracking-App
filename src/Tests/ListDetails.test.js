import Listitem from '../Components/Listitem';

describe('Tests: Listitem', () => {
  it('should return an action with type SORT_DATA', () => {
    const data = [
      {
        Country: 'Afghanistan',
        TotalConfirmed: 180520,
      },
    ];
    expect(Listitem(data)).toMatchSnapshot();
  });
});
