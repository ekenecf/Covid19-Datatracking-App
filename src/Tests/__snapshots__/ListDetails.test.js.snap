// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests: Listitem should return an action with type SORT_DATA 1`] = `
<React.Fragment>
  <div
    className="listItem"
  >
    <NavLink
      state={
        Object {
          "country": undefined,
        }
      }
      to="/Details"
    >
      <li
        className="countries"
      />
      <li
        className="confirmed"
      />
    </NavLink>
  </div>
</React.Fragment>
`;
